src/System.Diagnostics.Tracing.Logger/src/System/Diagnostics/Tracing/ILogger.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections.Generic;
+
+namespace System.Diagnostics.Tracing
+{
+    /// <summary>
+    /// A interface for logging.  It is supported by System.Diagnostics.Tracing.Logger, but it could in theory have
+    /// other implementations.   See Logger for more details on expected use.  
+    /// </summary>
+    public interface ILogger
+    {
+        /// <summary>
+        /// returns true if the given verbosity level is active (will be logged if 'Log' is called).  
+        /// </summary>
+        bool IsEnabled(LogLevel level);
+
+        /// <summary>
+        /// Log a single message.   A message has a name, a Verbosity level and a object that contains any other
+        /// information needed by the message.   Typical this object is created using an anonymous type.  
```

Comment:
In heavily instrumented apps, will this lead to the generation of a unique anonymous type for every structured log event? Since the `arguments` payload is the way to get event-specific data though the API, it seems like the number of types generated could be problematic. (Those types will also have to retain full metadata since the only way to retrieve their properties is via reflection - not sure how this fits with .NET Native.)